### DISCUSSION TITLE
Does the plugin work for GraphQL data?

### DISCUSSION THREAD

**Discussion by teinett** [ComfortablyCoding/strapi-plugin-transformer#13] [Q&A] [answered]
Does it work only in REST API? 
Or it works both in REST API and GraphQL?

**Comment by SoftCreatR** [comment]
Just had the same question. However, this doesn't seem to work for GraphQL yet :(

![image](https://user-images.githubusercontent.com/81188/154786595-9907c717-aa59-4103-9872-2d2b080e6f1a.png)

And omitting the property in the query:

![image](https://user-images.githubusercontent.com/81188/154786623-f3a5b406-cb2f-4cd7-a8ad-fdecc2bd4cde.png)

However: https://forum.strapi.io/t/discussion-regarding-the-complex-response-structure-for-rest-graphql-developer-experience/13400

**Comment by ComfortablyCoding** [comment, marked as answer]
Hello,

Currently this plugin does not work with GraphQL, but I am looking into adding support for it. 

I have managed to handle the response part and am currently investigating on how to modify the entity schemas to remove the attributes keys so no mismatch between the request and response.

**Reply by SoftCreatR** [reply]
Any progress here?

**Reply by ComfortablyCoding** [reply]
Unfortunately not. I was not able to find a reliably documented way of modifying the schema.

**Reply by 7alip** [reply]
Does it mean that if you will able to modify the schema, Graphql Codegen will generate the TypeScript types based on the modified schema?

**Reply by ComfortablyCoding** [reply]
Yes, that is the goal. The response and the request syntax should be the same. As mentioned above I was not able to find a suitable way to modify them.

**Reply by SalahAdDin** [reply]
As far as I remember, we can modify the schema manually, isn't it?

**Reply by HettieM** [reply]
Hi, have you had any further progress with graphQL modifications yet?

**Reply by SalahAdDin** [reply]
> Hi, have you had any further progress with graphQL modifications yet?

I don't think so.

**Reply by ComfortablyCoding** [reply]
Unfortunately no, does not look like it will be possible.
